### PR TITLE
feat: allow defining callbacks for confirm and cancel in custom modal…

### PR DESCRIPTION
**Pull request by JammingBen**
… components.

Follow-up after https://github.com/owncloud/web/pull/9967.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

### CODE REVIEW

**Review by dschmidt — APPROVED** [review]

File: packages/web-runtime/src/App.vue [hunk 1]
```diff
@@ -79,6 +80,32 @@ export default defineComponent({
 
     const activeRoute = computed(() => router.resolve(unref(router.currentRoute)))
 
+    const modalComponent = ref<VNodeRef>()
+    const onModalConfirm = () => {
+      if (store.state.modal.onConfirm) {
+        return store.state.onConfirm({ component: modalComponent })
```
[the remark this answers is not in the document]
Comment (JammingBen) [reply packages/web-runtime/src/App.vue:86]: Good catch 🙈

**Review by kulmann — APPROVED** [review]
👍

**Review by kulmann — APPROVED** [review]
again 👍 😁

File: packages/web-runtime/src/App.vue [hunk 2]
```diff
@@ -79,6 +80,33 @@ export default defineComponent({
 
     const activeRoute = computed(() => router.resolve(unref(router.currentRoute)))
 
+    const modalComponent = ref<VNodeRef>()
+    const onModalConfirm = (...args) => {
+      console.log(111, args)
```
Comment (dschmidt) [inline packages/web-runtime/src/App.vue:85]: 👀

Comment (kulmann) [reply packages/web-runtime/src/App.vue]: I've enabled auto-merge for this PR now. Just for the thrill of finding more dev leftovers and then having to be fast enough with pushing again 😁

**Pull request merged**